src/components/fileManaging/filesContainer.tsx

```diff
@@ -0,0 +1,251 @@
+    });
+  }
+
+  closePopUp = () => {
+    if (this.state.showPopup === true)
+      this.setState({
+        showPopup: false
+      });
+  }
+
+  handleImageChange(e: React.ChangeEvent<HTMLInputElement>) {
+    let file = e.target.files![0];
+    const extension = file.name.split('.')
+
+    this.setState({
+      currentFile: file,
+      added: false,
+      deleted: false
+    });
+
+    if (file.size > 1000000) {
+      this.setState({
+        sizeNotSupproted: true,
+        disabled: true
+      });
+    } else { this.setState({ sizeNotSupproted: false, }); }
+    if (extension[extension.length - 1] === 'pdf' || extension[extension.length - 1] === 'png' || extension[extension.length - 1] === 'jpeg') {
+      this.setState({
+        formatNotSupproted: false,
+        disabled: false,
+        sizeNotSupproted: false,
+      });
+    } else {
+      this.setState({
+        formatNotSupproted: true,
+        disabled: true
+      });
+    }
+  }
+
+  displayNumberOfMissingFiles(): JSX.Element {
+    const numberTotal = getBasicsAttachements().length - 1;
+    const numberAdded = this.state.filesAdded.length;
+    return (<p>{numberAdded} sur {numberTotal} fichiers fournis</p>);
+  }
+
+  handleTypeChange(attach_type: string) {
+    this.setState({
+      curentTypeFile: attach_type,
+      added: false,
+      deleted: false
+    });
+  }
+
+  async addNewFile(e: React.MouseEvent<HTMLButtonElement, MouseEvent>) {
+    e.preventDefault();
+    if (this.state.curentTypeFile === '' || this.state.currentFile === null) {
+      this.setState({ error: true });
+    } else {
+      let fileKey: string = '';
+      fileKey = await handleUpload(this.state.currentFile);
+      const newAttachement: IAttachement = { key: fileKey, attach_type: this.state.curentTypeFile, file: this.state.currentFile, fileName: this.state.currentFile.name }
+      this.removeElementInTypes();
+      const newFiles = [...this.state.filesAdded, newAttachement];
+      this.setState({
+        filesAdded: newFiles,
+        curentTypeFile: '',
+        currentFile: null,
+        error: false,
+        added: true
+      });
+      this.props.handleChangeAttachement(newFiles);
+    }
+  }
+
+  removeElementInTypes() {
+    const listTypes = this.state.typesList
+    const curentTypeFile = this.state.curentTypeFile;
+    const newList = listTypes.filter(function (type) {
+      return type.attach_type !== curentTypeFile;
+    });
+    this.setState({
+      typesList: newList
+    });
+  }
+
+  resetListOfTypes(typeFile: string) {
+    const type: IOption = { attach_type: typeFile, typeConverted: getTypeConverted(typeFile) }
+    this.setState(previousState => ({
+      typesList: [...previousState.typesList, type]
+    }));
+  }
+
+  removeElemFromListAdded(event: React.MouseEvent<HTMLButtonElement, MouseEvent>, file: IAttachement) {
+    event.preventDefault();
+
+    if (this.props.candId > 0) {
+      deleteAttachmentInApplication(this.props.candId, file.id!);
```

Comment:
There's a problem here. Let's take the following scenario : 
- I've already created the draft, and you're updating it (so `candId > 0`)
- I'm adding a new doc without saving the candidature (hence, the attachment is not save in backend) 
- I'm trying to delete the doc and you're asking the deletion in backend, where the attachment does not exists ... 

Let's discuss together for a solution